utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,551 @@
+    to be used to generate issue body for manual approval
+
+    :param license_verified: Whether the model is licensed under Apache 2.0
+    :type model_path: bool
+    :return: No return value expected
+    :rtype: None
+    """
+    try:
+        os.makedirs(OUTPUT_DIR, exist_ok=True)
+        license_var_filepath = OUTPUT_DIR + "/" + LICENSE_VAR_FILE
+        with open(license_var_filepath, "w") as f:
+            f.write(str(license_verified))
+    except Exception as e:
+        print(
+            f"Cannot store license_verified ({license_verified}) in {license_var_filepath}: {e}"
+        )
+
+
+def main(
+    model_id: str,
+    model_version: str,
+    tracing_format: str,
+    embedding_dimension: Optional[int] = None,
+    pooling_mode: Optional[str] = None,
+    model_description: Optional[str] = None,
+) -> None:
+    """
+    Perform model auto-tracing and prepare files for uploading to OpenSearch model hub
+
+    :param model_id: Model ID of the pretrained model
+    :type model_id: string
+    :param model_version: Version of the pretrained model for registration
+    :type model_version: string
+    :param tracing_format: Tracing format ("TORCH_SCRIPT", "ONNX", or "BOTH")
+    :type tracing_format: string
+    :param embedding_dimension: Embedding dimension input
+    :type embedding_dimension: int
+    :param pooling_mode: Pooling mode input ("CLS", "MEAN", "MAX", "MEAN_SQRT_LEN" or None)
+    :type pooling_mode: string
+    :param model_description: Model description input
+    :type model_description: string
+    :return: No return value expected
+    :rtype: None
+    """
+
+    print("\n=== Begin running model_autotracing.py ===")
+    print("Model ID: ", model_id)
+    print("Model Version: ", model_version)
+    print("Tracing Format: ", tracing_format)
+    print(
+        "Embedding Dimension: ",
+        embedding_dimension if embedding_dimension is not None else "Default",
+    )
+    print("Pooling Mode: ", pooling_mode if pooling_mode is not None else "Default")
+    print(
+        "Model Description: ",
+        model_description if model_description is not None else "Default",
+    )
+    print("==========================================")
+
+    ml_client = MLCommonClient(OPENSEARCH_TEST_CLIENT)
+
+    pre_trained_model = SentenceTransformer(model_id)
+    original_embedding_data = list(
+        pre_trained_model.encode(TEST_SENTENCES, convert_to_numpy=True)
+    )
+
+    pre_trained_model.save(path=TEMP_MODEL_PATH)
+    license_verified = verify_license_in_md_file()
+    try:
+        shutil.rmtree(TEMP_MODEL_PATH)
+    except Exception as e:
+        assert False, f"Raised Exception while deleting {TEMP_MODEL_PATH}: {e}"
+
+    if tracing_format in [TORCH_SCRIPT_FORMAT, BOTH_FORMAT]:
+        print("--- Begin tracing a model in TORCH_SCRIPT ---")
+        (
+            torchscript_model_path,
+            torchscript_model_config_path,
+        ) = trace_sentence_transformer_model(
```

Comment:
We are also initiating `SentenceTransformer` class again inside this `trace_sentence_transformer_model` method. Can't we send `pre_trained_model` as parameter so that we don't need to initialize twice?

Reply:
We have to initiate `SentenceTransformer` twice for the case where we trace both formats anyway because the folder path of the torch_script and onnx should be different. So, I decided to initiate a new one in all cases.